### ISSUE TITLE
WhatWeb crashes out on bad redirect

### ISSUE THREAD

**Issue by rhodejo** [urbanadventurer/WhatWeb#252]
IP[x.x.x.x], RedirectLocation[https://x.x.x.x:81:743/], Title[301 Moved Permanently]
/usr/lib/ruby/2.3.0/uri/generic.rb:1102:in `rescue in merge': bad URI(is not URI?): https://x.x.x.x:81:743/ (URI::InvalidURIError)

A bad redirect crashes WhatWeb.

**Comment by jas502n** [comment]
```
/usr/lib/ruby/2.3.0/uri/generic.rb:1102:in `rescue in merge': `bad URI(is not URI?):  `http://www.xxxx.cn/download.jsp (URI::InvalidURIError)
        from /usr/lib/ruby/2.3.0/uri/generic.rb:1099:in `merge'
        from /usr/lib/ruby/2.3.0/uri/rfc3986_parser.rb:89:in `each'
        from /usr/lib/ruby/2.3.0/uri/rfc3986_parser.rb:89:in `inject'
        from /usr/lib/ruby/2.3.0/uri/rfc3986_parser.rb:89:in `join'
        from /usr/lib/ruby/2.3.0/uri/common.rb:265:in `join'
        from /usr/share/whatweb/lib/target.rb:237:in `get_redirection_target'
        from /usr/bin/whatweb:997:in `block (3 levels) in <main>'
        from /usr/bin/whatweb:980:in `loop'
        from /usr/bin/whatweb:980:in `block (2 levels) in <main>'
root@kali:/tmp#

```

**Comment by urbanadventurer** [comment]
Thanks @rhodejo and @jas502n for reporting this issue.
It's fixed in a6cdfa87d478915e10e5f1b9213c259567c75b65 in 0.5.0-dev branch
